### PR TITLE
Fixes an XSS issue and removes the stacktrace to avoid leaking server information when an unrecoverable error occurs

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Fixes an XSS issue and removes the stacktrace to avoid leaking server information when an unrecoverable error occurs
## Impacts

Impacts any part of the portal that would expect multiple parameters (Non seen).
## Delete branch after merge

NO

### CODE REVIEW

**Comment by extrafu** [comment]
That has to go in v4.5.2 too.

**Comment by jrouzierinverse** [comment]
Review please

File: html/captive-portal/lib/captiveportal/Role/Request.pm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    my ($self,$param) = @_;
+    return encode_entities($self->param($param));
+}
+
+around param => sub {
+    my ($orig,$self,@args) = @_;
+    return @args ? scalar $self->$orig(@args) : undef;
+};
+
+sub param_old {
```
Comment (cgx) [inline html/captive-portal/lib/captiveportal/Role/Request.pm]: Do we really need to keep this method?

File: html/captive-portal/lib/captiveportal/Role/Request.pm [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+}
+ 
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2014 Inverse inc.
```
Comment (cgx) [inline html/captive-portal/lib/captiveportal/Role/Request.pm]: 2015

File: NEWS.asciidoc [hunk 3]
```diff
@@ -53,6 +53,7 @@ Bug Fixes
 * Fix wrong regex to detect ifindex in Cisco switches
 * Honor order of profiles when matching profile filters
 * Fixed URI based portal profiles
+* Removed XSS error in the portal
```
Comment (cgx) [inline NEWS.asciidoc]: I would rephrase this: _Fixed XSS vulnerabilities in the portal_.

File: html/captive-portal/lib/captiveportal/Role/Request.pm (hunk 1 above)
Comment (rouzier) [inline html/captive-portal/lib/captiveportal/Role/Request.pm]: No we do not need to keep it.

**Comment by jrouzierinverse** [comment]
Please review

**Comment by jrouzierinverse** [comment]
Please review

**Pull request merged**